File: Examples/test-suite/perl5/overload_extend_runme.pl
use strict;
use warnings;
use Test::More tests => 7;
BEGIN { use_ok 'overload_extend' }
require_ok 'overload_extend';

my $f = overload_extend::Foo->new();
is($f->test, 0, 'f.test()');
is($f->test(3), 1, 'f.test(3)');
is($f->test("hello"), 2, 'f.test("hello")');
is($f->test(3, 2), 5, 'f.test(3, 2)');
is($f->test(3.0), 1003, 'f.test(3.0)');

File: Examples/test-suite/php/overload_extend_runme.php
<?php
require "tests.php";

$f = new Foo();
check::equal($f->test(), 0, '$f->test()');
check::equal($f->test(3), 1, '$f->test(3)');
check::equal($f->test("hello"), 2, '$f->test("hello")');
check::equal($f->test(3.0, 2.0), 5.0, '$f->test(3.0, 2.0)');
check::equal($f->test(3.0), 1003.0, '$f->test(3.0)');

check::done();
